packages/core/src/test/shared/sam/utils.test.ts

```diff
@@ -148,4 +163,113 @@ describe('SAM utils', async function () {
+                assert(telemetryStub.notCalled)
+            }
+        })
+
+        it('throws VersionTooLow error when SAM CLI version is below 1.53.0', async () => {
+            const lowVersion = '1.52.0'
+            getOrDetectSamCliStub.resolves({ path: '/usr/local/bin/sam' })
+            executeStub.resolves({ version: lowVersion })
+
+            try {
+                await getSamCliPathAndVersion()
+                assert.fail('Should have thrown an error')
+            } catch (error) {
+                assert(error instanceof ToolkitError)
+                assert.strictEqual(error.code, 'VersionTooLow')
+                assert.strictEqual(error.message, 'SAM CLI version 1.53.0 or higher is required')
+                // records telemetry even when version is too low
+                assert(telemetryStub.calledOnceWith({ version: lowVersion }))
+            }
+        })
+    })
+
+    describe('get/update recent response', () => {
+        // Create stub for globals.context.workspaceState.get
+        const mementoRootKey = 'samcli.utils.key'
+        const nonExistingMementoRootKey = 'samcli.utils.no.key'
+        const identifier = 'us-east-1'
+        const key1 = 'stackName'
+        const key2 = 'bucketName'
+        const value1 = 'myStackName'
+        const value2 = 'myBucketName'
+
+        it('1. getRecentResponse should return undefined when mementoRootKey does not exist', async () => {
+            assert(!getRecentResponse(nonExistingMementoRootKey, identifier, key1))
+        })
+
+        it('2. updateRecentResponse should return the undefined when mementoRootKey does not exist', async () => {
+            try {
+                await updateRecentResponse(mementoRootKey, identifier, key1, value1)
+            } catch (err) {
+                assert.fail('Should have succeed')
+            }
+        })
+
+        it('3. getRecentResponse should return the correct value', async () => {
+            const result = getRecentResponse(mementoRootKey, identifier, key1)
+            assert.strictEqual(result, value1)
+        })
+
+        it('4. updateRecentResponse should return the update only specified key', async () => {
+            await updateRecentResponse(mementoRootKey, identifier, key2, value2)
+            const result1 = getRecentResponse(mementoRootKey, identifier, key1)
+            const result2 = getRecentResponse(mementoRootKey, identifier, key2)
+            assert.strictEqual(result1, value1)
+            assert.strictEqual(result2, value2)
+        })
+
+        it('5. updateRecentResponse should log when failure to update value', async () => {
+            sinon.stub(globals.context.workspaceState, 'update').rejects(new Error('Error updating value'))
+            try {
+                await updateRecentResponse(mementoRootKey, identifier, key2, value2)
+            } catch (err) {
+                assert.fail('Should have swallowed exception')
```

Comment:
It seems you want this to succeed, but assert fail is added in try catch. Does it mean either this success or failed, test should pass?